components/org.wso2.carbon.identity.scim2.common/src/main/java/org/wso2/carbon/identity/scim2/common/impl/SCIMUserManager.java

```diff
@@ -486,88 +486,361 @@ public User updateUser(User user, Map<String, Boolean> requiredAttributes) throw
+            domainsOfUserStores.add(domainName);
         }
+        return domainsOfUserStores.toArray(new String[0]);
+    }
+
+    /**
+     * Method to filter users if the user store is not an instance of PaginatedUserStoreManager and
+     * ENABLE_PAGINATED_USER_STORE is not enabled.
+     *
+     * @param node   Expression node
+     * @param limit  Number of users required for counting
+     * @param offset Starting user index for start counting
+     * @return List of paginated set of users.
+     * @throws NotImplementedException Not supported filter operation
+     * @throws UserStoreException
+     */
+    private String[] filterUsersUsingLegacyAPIs(ExpressionNode node, int limit, int offset)
+            throws NotImplementedException, CharonException {
+
+        String[] userNames;
+
+        // Set filter values.
+        String attributeName = node.getAttributeValue();
+        String filterOperation = node.getOperation();
+        String attributeValue = node.getValue();
+
+        try {
+            if (SCIMConstants.UserSchemaConstants.GROUP_URI.equals(attributeName)) {
+                if (carbonUM instanceof AbstractUserStoreManager) {
+                    String[] roleNames = getRoleNames(filterOperation, attributeValue);
+                    userNames = getUserListOfRoles(roleNames);
+                } else {
+                    String errorMessage = String
+                            .format("Filter operator %s is not supported by the user store.", filterOperation);
+                    throw new NotImplementedException(errorMessage);
+                }
+            } else
+                // Get the user name of the user with this id.
+                userNames = getUserNames(attributeName, filterOperation, attributeValue);
+        } catch (UserStoreException e) {
+            String errorMessage = "Error while filtering the users";
+            throw new CharonException(errorMessage, e);
+        }
+
+        userNames = paginateUsers(userNames, limit, offset);
+        return userNames;
+    }
+
+    /**
+     * Method to remove duplicate users and get the user details.
+     *
+     * @param userNames          Filtered user names
+     * @param requiredAttributes Required attributes in the response
+     * @return Users list with populated attributes
+     * @throws CharonException Error in retrieving user details
+     */
+    private List<Object> getDetailedUsers(String[] userNames, Map<String, Boolean> requiredAttributes)
+            throws CharonException {
+
+        List<Object> filteredUsers = new ArrayList<>();
+        // 0th index is to store total number of results.
+        filteredUsers.add(0);
+
+        // Remove duplicate users.
+        HashSet<String> userNamesSet = new HashSet<>(Arrays.asList(userNames));
+        userNames = userNamesSet.toArray(new String[0]);
+        filteredUsers.set(0, userNames.length);     //number of filtered users
```

Comment:
Shall we make this comment in formal way in a new line?

Reply:
This issue does not exist in the latest code